usr/share/rear/output/PXE/default/800_copy_to_tftp.sh

```diff
@@ -32,14 +32,17 @@ else
     [[ ! -d "$PXE_TFTP_LOCAL_PATH" ]] && mkdir $v -m 750 "$PXE_TFTP_LOCAL_PATH" >&2
 fi
 
-
-cp -pL $v "$KERNEL_FILE" "$PXE_TFTP_LOCAL_PATH/$PXE_KERNEL" || Error "Failed to copy KERNEL_FILE '$KERNEL_FILE'"
-cp -a $v "$TMP_DIR/$REAR_INITRD_FILENAME" "$PXE_TFTP_LOCAL_PATH/$PXE_INITRD" || Error "Failed to copy initrd '$REAR_INITRD_FILENAME'"
+# Follow symbolic links to ensure the real content gets copied
+# but do not preserve mode,ownership,timestamps (i.e. no -p option) because that may fail (on sshfs) like
+# "cp: failed to preserve ownership for '/tmp/rear-efi.XXXXXXXXXX/EFI/BOOT/kernel': Operation not permitted"
+cp -L $v "$KERNEL_FILE" "$PXE_TFTP_LOCAL_PATH/$PXE_KERNEL" || Error "Failed to copy KERNEL_FILE '$KERNEL_FILE'"
+cp -L $v "$TMP_DIR/$REAR_INITRD_FILENAME" "$PXE_TFTP_LOCAL_PATH/$PXE_INITRD" || Error "Failed to copy initrd '$REAR_INITRD_FILENAME'"
 echo "$VERSION_INFO" >"$PXE_TFTP_LOCAL_PATH/$PXE_MESSAGE"
 # files must be readable for others for PXE
-chmod 444 "$PXE_TFTP_LOCAL_PATH/$PXE_KERNEL"
-chmod 444 "$PXE_TFTP_LOCAL_PATH/$PXE_INITRD"
-chmod 444 "$PXE_TFTP_LOCAL_PATH/$PXE_MESSAGE"
+# files should be writebale by owner or overwriting it on later runs will fail
```

Comment:
Probably would be good to add comment that files should be writeable by owner for overwriting/deleting and readable by world for tftp server to read them. As the initrd can contain sensitive data we might think about making this behaviour a configuration item in the future.

Reply:
So the commet in there is not clear enough?

Reply:
Read for others should not be needed. depending on your tftp server and owner/group it should be fine with 640.
I don't see this as a issue as long as you do not copy shadow/passwd by rear config. Also I control access with a different method - basically over DHCP/tftp only listening on one interface and for specific MAC addresses.

Reply:
For comparison what owner and group is used and
what permissions are set on the NFS server with
```
root@nfs_server # cat /etc/exports
/nfs    *(rw,root_squash,sync,no_subtree_check)
```
when on 'localhost'
```
OUTPUT=ISO
BACKUP=NETFS
BACKUP_OPTIONS="nfsvers=3,nolock"
BACKUP_URL=nfs://192.168.122.1/nfs
```
is used after "rear mkbackup"
```
root@nfs_server # ls -lhd /nfs/localhost
drwxr-x--- 2 nobody nobody 4.0K Dec  8 12:37 /nfs/localhost

root@nfs_server # ls -lhtr /nfs/localhost
-rw------- 1 nobody nobody  71M Dec  8 12:32 rear-localhost.iso
-rw------- 1 nobody nobody  266 Dec  8 12:32 VERSION
-rw------- 1 nobody nobody  202 Dec  8 12:32 README
-rw------- 1 nobody nobody 1.8M Dec  8 12:32 rear-localhost.log
-rw------- 1 nobody nobody 1.3G Dec  8 12:37 backup.tar.gz
-rw------- 1 nobody nobody 6.0M Dec  8 12:37 backup.log
```
So perhaps access mode 640 for PXE should be preferred?

I am not a PXE user but I think accessability of a PXE server
where ReaR stores kernel and recovery system initrd
can be considered similar as accessability of a NFS server
where ReaR stores kernel, recovery system initrd and backup.

I think both can be assumed to be internal servers of a company or organization
where the admins of those servers and their environment have set up
appropriate general access restrictions for their servers
so that file and directory permissions of ReaR's files
are not really critical settings.

E.g. I assume when someone can write things on a PXE server
he can determine what other systems boot so he has similar power
as a user who has physical access to a computer hardware.

Furthermore according to
https://en.wikipedia.org/wiki/Trivial_File_Transfer_Protocol#Security_considerations
`TFTP includes no login or access control mechanisms.`
so the whole PXE environment must be isolated from foreign access
and it seems within the PXE environment all things are basically trusted
so file and directory permissions of ReaR's files are not really critical settings.

Again: I am not a PXE user so my above assumtions could be plain wrong.